src/common/k8s-api/api-manager.ts

```diff
@@ -6,15 +6,24 @@
 import type { KubeObjectStore } from "./kube-object.store";
 
 import { action, observable, makeObservable } from "mobx";
-import { autoBind, iter } from "../utils";
+import { autoBind, isDefined, iter } from "../utils";
 import type { KubeApi } from "./kube-api";
-import type { KubeObject } from "./kube-object";
-import type { IKubeObjectRef } from "./kube-api-parse";
+import type { KubeJsonApiDataFor, KubeObject, ObjectReference } from "./kube-object";
 import { parseKubeApi, createKubeApiURL } from "./kube-api-parse";
 
+export type RegisterableStore<S> = S extends KubeObjectStore<any, any, any>
+  ? S
+  : never;
+export type RegisterableApi<A> = A extends KubeApi<any, any>
+  ? A
+  : never;
+export type KubeObjectStoreFrom<A> = A extends KubeApi<infer K, infer D>
```

Comment:
`S`, `A`, `D`, `K` doesn't mean anything to developer who is not familiar with the implementation of e.g. `KubeApi`.

`Store`, `Api`, `????`, `KubeObject` (??) would increase the readability. 

We don't use single letter variable naming in implementation, so why we should do it in typing? We should stop using it when it's not absolute clear.